c++/configure.ac

```diff
@@ -52,6 +52,19 @@ AC_PROG_CXX
 AC_LANG([C++])
 AX_CXX_COMPILE_STDCXX_11
 
+AC_C_BIGENDIAN([CXXFLAGS="$CXXFLAGS -D_KJ_BIGENDIAN=1"],
+               [CXXFLAGS="$CXXFLAGS -D_KJ_BIGENDIAN=0"],
+               [AC_MSG_WARN([could not determine system endianness: assuming little endian])
+                CXXFLAGS="$CXXFLAGS -D_KJ_BIGENDIAN=0"])
```

Comment:
The `AC_MSG_WARN` feels like something of a cop out, but I found it hard to justify breaking the build over something that only really seems to affect a small set of features.